plugins/inputs/promql/query.go

```diff
@@ -0,0 +1,157 @@
+package promql
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	apiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+)
+
+type query struct {
+	Name  string `toml:"name"`
+	Query string `toml:"query"`
+	Limit uint64 `toml:"limit"`
+
+	client  *client
+	options []apiv1.Option
+	log     telegraf.Logger
+}
+
+func (q *query) init(c *client, log telegraf.Logger, options ...apiv1.Option) {
+	q.client = c
+	q.options = append(options, apiv1.WithLimit(q.Limit))
+	q.log = log
+}
+
+type InstantQuery struct {
+	query
+}
+
+func (q *InstantQuery) init(c *client, log telegraf.Logger, options ...apiv1.Option) {
+	q.query.init(c, log, options...)
+}
+
+func (q *InstantQuery) execute(ctx context.Context, acc telegraf.Accumulator, t time.Time) error {
+	results, warnings, err := q.client.Query(ctx, q.Query, t, q.options...)
+	if err != nil {
+		return fmt.Errorf("querying %q failed: %w", q.client.url, err)
```

Comment:
Does this expose the internal URL in the error? Should we use a more generic message, like "query failed," to avoid leaking internal details?

Reply:
Which internal details do you mean? The URL should not contain any credentials but just the "url text" used in the config.

Reply:
That’s ok, I just wanted to confirm. Thanks for clarifying!